src/Piral.Blazor.Tools/content/src/blazor.codegen

```diff
@@ -293,16 +294,33 @@ function toFramework(files) {
 }
 async function prepare(targetDir, staticAssets) {
   const piralPiletFolder = (0, import_path5.resolve)(__dirname, "..");
-  const project = require((0, import_path5.resolve)(piralPiletFolder, pjson));
-  const appdir = findAppDir(piralPiletFolder, project.piral.name);
+  const packageJson = require((0, import_path5.resolve)(piralPiletFolder, packageJsonFilename));
+
+  const piletJsonFilePath = (0, import_path5.join)(piralPiletFolder, piletJsonFilename).replace(/\\/g, "/");
+  const piletJsonFileExists = (0, import_fs5.existsSync)(piletJsonFilePath);
+  let instanceName;
+  if(piletJsonFileExists){
+    const piletJson = require((0, import_path5.resolve)(piralPiletFolder, piletJsonFilename));
+    const selectedInstance = Object.keys(piletJson.piralInstances).find(key => piletJson.piralInstances[key].selected);
+    if(selectedInstance !== undefined){
+      instanceName = selectedInstance;
+    } else{
+      instanceName = Object.keys(piletJson.piralInstances)[0];
+    }
+  } else{
+    instanceName = packageJson.piral.name;
+  }
+  
+
+  const appdir = findAppDir(piralPiletFolder, instanceName);
   const manifestSource = staticAssets.Assets.find((m) => m.AssetTraitName === "BlazorWebAssemblyResource" && m.AssetTraitValue === "manifest" && m.RelativePath.endsWith(bbjson));
   if (!manifestSource) {
     throw new Error(`Could not find the "${bbjson}" in ${swajson}. Something seems to be wrong.`);
   }
   const appFrameworkDir = (0, import_path5.resolve)(appdir, "app", "_framework");
   const bbAppShellPath = (0, import_path5.resolve)(appFrameworkDir, bbjson);
   const blazorInAppshell = (0, import_fs4.existsSync)(bbAppShellPath);
-  const shellPackagePath = (0, import_path5.resolve)(appdir, pjson);
+  const shellPackagePath = (0, import_path5.resolve)(appdir, packageJsonFilename);
```

Comment:
Danger - the whole file is generated.

If you want to change the contents of the file you should change the sources: https://github.com/smapiot/Piral.Blazor/tree/blazor-6.0/src/blazor-codegen/src

For instance, the `prepare` is in https://github.com/smapiot/Piral.Blazor/blob/blazor-6.0/src/blazor-codegen/src/prepare.ts